RecoMET/METFilters/plugins/BadChargedCandidateFilter.cc

```diff
@@ -0,0 +1,159 @@
+
+//
+// class declaration
+//
+
+class BadChargedCandidateFilter : public edm::global::EDFilter<> {
+public:
+  explicit BadChargedCandidateFilter(const edm::ParameterSet&);
+  ~BadChargedCandidateFilter();
+
+private:
+  virtual bool filter(edm::StreamID iID, edm::Event&, const edm::EventSetup&) const override;
+
+      // ----------member data ---------------------------
+
+  edm::EDGetTokenT<edm::View<reco::Candidate> >   tokenPFCandidates_;
+  edm::EDGetTokenT<edm::View<reco::Muon> >   tokenMuons_;
+
+  const bool taggingMode_;
+  const bool debug_;
+  const double          maxDR_;
+  const double          minPtDiffRel_;
+  const double          minMuonTrackRelErr_;
+  const double          innerTrackRelErr_;
+  const double          minMuonPt_;
+  const double          segmentCompatibility_;
+
+};
+
+//
+// constructors and destructor
+//
+BadChargedCandidateFilter::BadChargedCandidateFilter(const edm::ParameterSet& iConfig)
+  : tokenPFCandidates_ ( consumes<edm::View<reco::Candidate> >(iConfig.getParameter<edm::InputTag> ("PFCandidates")  ))
+  , tokenMuons_ ( consumes<edm::View<reco::Muon> >(iConfig.getParameter<edm::InputTag> ("muons")  ))
+  , taggingMode_          ( iConfig.getParameter<bool>    ("taggingMode") )
+  , debug_                ( iConfig.getParameter<bool>    ("debug") )
+  , maxDR_                ( iConfig.getParameter<double>  ("maxDR") )
+  , minPtDiffRel_         ( iConfig.getParameter<double>  ("minPtDiffRel") )
+  , minMuonTrackRelErr_   ( iConfig.getParameter<double>  ("minMuonTrackRelErr") )
+  , innerTrackRelErr_     ( iConfig.getParameter<double>  ("innerTrackRelErr") )
+  , minMuonPt_            ( iConfig.getParameter<double>  ("minMuonPt") )
+  , segmentCompatibility_ ( iConfig.getParameter<double>  ("segmentCompatibility") )
+{
+  produces<bool>();
+}
+
+BadChargedCandidateFilter::~BadChargedCandidateFilter() { }
+
+
+//
+// member functions
+//
+
+// ------------ method called on each new Event  ------------
+bool
+BadChargedCandidateFilter::filter(edm::StreamID iID, edm::Event& iEvent, const edm::EventSetup& iSetup) const
+{
+  using namespace std;
+  using namespace edm;
+
+  typedef View<reco::Candidate> CandidateView;
+  Handle<CandidateView> pfCandidates;
+  iEvent.getByToken(tokenPFCandidates_,pfCandidates);
+
+  typedef View<reco::Muon> MuonView;
+  Handle<MuonView> muons;
+  iEvent.getByToken(tokenMuons_,muons);
+
+  bool foundBadChargedCandidate = false;
+
+  for ( unsigned i=0; i < muons->size(); ++i ) { // loop over all muons
+
+    const reco::Muon & muon = (*muons)[i];
+
+    reco::TrackRef bestMuonTrack = muon.muonBestTrack();
+
+    if (debug_) cout<<"BadChargedCandidate test:Muon "<< i << endl;
+    // reco::TrackRef innerMuonTrack = muon.innerTrack();
+    //  if ( muon.pt() < minMuonPt_ && innerMuonTrack->pt() < minMuonPt_) {
+    //  if (debug_) cout <<"skipping the muon because low muon pt" << endl;
+    //  continue ; } {
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant